source/simulator/assembly.cc

```diff
@@ -1419,7 +1419,8 @@ namespace aspect
                    * on the direction of the flow. Thus the flow-dependent terms below are
                    * only calculated if the edge is an inflow edge.
                    */
-                  const bool inflow = ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.);
+                  const bool inflow = (advection_field.is_temperature()
+                                       || ((current_u * scratch.face_finite_element_values->normal_vector(q)) < 0.));
```

Comment:
I just thought this from PDE. The temperature equation is a second order equation in spatial, so we could impose entire Dirichlet boundary condition, but compositional equation is a first order equation in spatial, so we can only impose the inflow Dirichlet boundary condition. However, the current code seems only impose inflow boundary condition, then for temperature which may have also imposed outflow boundary condition (which I am not sure if physically practical), the code will impose nothing on the outflow boundary, which then becomes a weakly imposed zero Neumman boundary condition.